src/settings.ts

```diff
@@ -32,33 +34,69 @@ export class RDSettingTab extends PluginSettingTab {
 		let logoutButton: ButtonComponent;
 		new Setting(containerEl)
 			.setName('Login')
-			.setDesc('Login and get a token')
+			.setDesc('Login via OAuth. Attempts deprecated username/password login if oAuth fails')
 			.addButton((btn) => {
 				loginButton = btn;
 				btn
-					.setButtonText(loggedIn ? `Logged in as ${this.plugin.settings.username}` : 'Login')
+					.setButtonText(loggedIn ? `Registered "${client_name}" application` : 'Login')
 					.setDisabled(loggedIn)
 					.setCta()
 					.onClick(async () => {
-						// Do login
-						new LoginModal(this.app, (username, password) => {
-							this.plugin.api.getToken(username, password)
-								.then(async (token) => {
-									// update values
-									this.plugin.settings.apiToken = token;
-									this.plugin.settings.username = username;
-									await this.plugin.saveSettings();
-									// update ui
-									loginButton.setButtonText(`Logged in as ${this.plugin.settings.username}`);
-									loginButton.setDisabled(true);
-									logoutButton.setDisabled(false);
-								}).catch((error) => {
-									console.log("Login error", error);
-									new Notice('Login error, check your credentials');
-								});
-						}).open();
+						loginButton.setDisabled(true);
+						let cancelled = false;
+						try {
```

Comment:
Instead of this, you can first query `/api/info`. On recent Readeck versions, you'll get something looking like this:

```json
{
  "version": {
    "canonical": "0.21",
    "release": "0.21.4",
    "build": ""
  },
  "features": [
    "email",
    "oauth"
  ]
}
```

If `oauth` is in `features` then OAuth is supported. Otherwise (including 404 on this route), you can use the legacy login for as long as you want.

https://codeberg.org/readeck/browser-extension/src/commit/de98607d9f76d2422dd27c3d3be0f99b704a2863/src/lib/readeck-api.js#L17-L36

https://codeberg.org/readeck/browser-extension/src/commit/de98607d9f76d2422dd27c3d3be0f99b704a2863/src/app/login.svelte#L37-L38